### ISSUE TITLE
[Bug]: Can not fire click(or double click) event if set a '-webkit-app-region: drag;' style

### ISSUE THREAD

**Issue by kyle-go** [electron/electron#37789]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

23.2.1

### What operating system are you using?

macOS

### Operating System Version

13.3 (22E252)

### What arch are you using?

x64

### Last Known Working Electron version

24

### Expected Behavior

I can drag the item to move window and fire click (double click) event.

### Actual Behavior

click event NOT FIRED.

### Testcase Gist URL

https://gist.github.com/fcc8358e36cae1dc3ed07458e4be46dc

### Additional Information

Is it a bug?

**Comment by kyle-go** [comment]
Actually, as the window is a small tips window, I have to set the window property of :

``` 
    resizable: true,
    minimizable: false,
    maximizable: false,
    acceptFirstMouse: true,
    // movable: true,
    alwaysOnTop: true,
    fullscreenable: false,
    frame: false,
```

**Comment by jordandearsley** [comment]
Yep, all mouse events aren't being picked up when `-webkit-app-region: drag;`

**Comment by kyle-w-20230331** [comment]
any update?  thanks

**Comment by pacocoursey** [comment]
Seeing the same broken behavior, I'm guessing caused by #36230. It only applies to frameless windows (`frame: false` or `titleBarStyle: 'hidden'`). A workaround is to apply `-webkit-app-region: no-drag` to children that should receive pointer events.

Elements with `-webkit-app-region: drag` seem to act as `z-index: Infinity`, blocking all children from interaction and even becoming un-selectable via Chrome DevTools, which can make this really hard to debug!

The expected behavior is to follow CSS layering order, ignoring if `-webkit-app-region` is set.

**Comment by pacocoursey** [comment]
@codebytere Would you advocate for this to be prioritized? It's a pretty huge bug that makes any elements using `-webkit-app-region` completely unclickable and un-interactable, breaking UI across the majority of web apps that use Electron.

**Comment by codebytere** [comment]
@pacocoursey yes - it's on my personal docket and being actively worked on :)

**Comment by gerhardberger** [comment]
Put up a fix for this issue here: https://github.com/electron/electron/pull/38208

**Comment by kyle-w-20230331** [comment]
any process？

**Comment by liees** [comment]
The mouse can trigger the pointer event normally, but my handle is completely useless. The magic is that I use git bash to start the development mode of electron and I can use the handle to click

**Comment by kashamalasha** [comment]
On Windows behavior is the same. You can not handle mouse events without explicitly setting -webkit-app-region: drag in CSS on the element.

**Comment by kyle-w-20230331** [comment]
@codebytere any process? thanks

**Comment by foxt** [comment]
Noticed this regression when upgrading our app from v19 to latest.

@pacocoursey -- ideally, the expected behaviour is the old behaviour where any region of the page tagged with -war:d would become draggable

**Comment by jtvberg** [comment]
Sorry, not following. It looks like the bug was identified and a fix was found but the deployment of that fix was delayed and then closed out. What is the status of this bug? I still see the same issue in 26.4.0

**Comment by sdr0x07b6** [comment]
I have also been in trouble for a long time.

https://github.com/electron/electron/pull/38208
It seems to be showing that there are conflicts in the modified branch.
If the inability to fix the conflicts is the cause of the neglect, I hope it will be reopened soon, as the conflicts are likely to grow even larger the longer they are neglected.

**Comment by jtbandes** [comment]
Hi there – wondering if there is any update to share on this issue? It seems like this bug makes it impossible to implement "double-click to maximize" behavior on linux (although for some reason, [triple-click does work](https://github.com/microsoft/vscode/issues/43154#issuecomment-1206266360)) if you want to be able to double-click a draggable region (which would be common for custom title bars).

**Comment by sdr0x07b6** [comment]
@jtbandes Hi, I wanted to know the current status too, so I asked them directly.
https://github.com/electron/electron/pull/38208#issuecomment-1820432052
To be honest, I wasn't sure, but apparently some are of the opinion that this is not a bug. It seems that this is why progress is being hindered. In other words, the code has already been fixed, but the merging of the code seems to have been daringly stopped.
As for me, I am convinced that it is a bug.

**Comment by kyle-go** [comment]
It's definitely a bug, it's not solved yet

**Comment by sdr0x07b6** [comment]
bump

**Comment by KochiyaOcean** [comment]
I've noticed that in the [Breaking changes document](https://github.com/electron/electron/blob/3eb94b72ba6659fb27d59a71f1f9ba945219bd15/docs/breaking-changes.md#planned-breaking-api-changes-230), it mention that since Electron@23, the behavior of `-webkit-app-region` on macOS platform will matches that on Windows / Linux. Since this issue also happens since Electron@23, it looks like related this breaking change.

**Comment by sdr0x07b6** [comment]
This issue is not good with multiple threads..

Well, I have a new suggestion on this.
https://github.com/electron/electron/pull/38208#issuecomment-2144092250

Simply claiming this issue is a bug will not save us from our pain.
What do you guys think about accepting the new decision on the Electron side, but hoping that the new way of specifying our slightly unique UX-first requirements will be implemented as yet another intention?
I think it would be more acceptable than simply continuing to call the specification change a bug.

I think there are better ideas than what I'm proposing. Let's all get our voices heard by Electorn.

**Comment by bpasero** [comment]
@codebytere this is an issue that bugs VS Code, where right-clicking into the custom title bar does not bring up the context menu to configure the title bar anymore since a long time. Is it possible to look at the provided PR to see if it fixes the issue?

**Comment by codebytere** [comment]
@bpasero the larger issue here is that it was never intended to work this way. It did work this way, essentially by accident, for several versions - some people came to rely on the functionality while it was briefly available, but switching it back would also break a lot of existing apps. That's why it remains stalled - we're still considering how to move forward or if it's the right choice to do so. Gerhard's PR restores the temporary functionality, but does so by doing what we also didn't want to do - going against the way draggable regions were intended to work in Chromium.

**Comment by bpasero** [comment]
@codebytere but there must be something Chrome is doing differently then because I have a draggable region at the empty space after tabs and it supports a context menu:

![Image](https://github.com/user-attachments/assets/2dc434e5-35d5-40fe-b894-3bdcb59d7804)

**Comment by codebytere** [comment]
@bpasero this is specific to frameless windows - a window with a custom `titleBarOverlay` is technically frameless. Chrome's window by comparison isn't frameless and wouldn't hit this.

**Comment by codebytere** [comment]
@bpasero would you mind opening a new issue for the `contextmenu` event bug?

**Comment by bpasero** [comment]
@codebytere https://github.com/electron/electron/issues/44386

Just to clarify: no other mouse event would work other than `context-menu`? Is there a reason? I now see we have code to detect a `Cmd-click` on macOS too.

**Comment by gerhardberger** [comment]
for visibility, I published an NPM module, [`electron-drag-click`](https://github.com/gerhardberger/electron-drag-click),  that achieves the same functionality I put up in my [initial PR](https://github.com/electron/electron/pull/38208). This way somebody wishing this behavior can add it.

**Comment by noobfromph** [comment]
For windows and linux users, you can use this plugin
[electron-click-drag-plugin](https://github.com/noobfromph/electron-click-drag-plugin)

```
const { app, BrowserWindow, ipcMain } = require('electron');
const dragAddon = require('electron-click-drag-plugin'); // Loads the native addon

app.whenReady().then(() => {
  const win = new BrowserWindow({
    width: 800,
    height: 800,
    frame: false, // Important: no OS window frame
    webPreferences: {
      nodeIntegration: true,
      contextIsolation: false, // only for test/dev
      devTools: true
    },
  });

  win.loadFile('index.html');
  win.webContents.openDevTools();

  ipcMain.on('start-drag', () => {
    try {
      const hwndBuffer = win.getNativeWindowHandle();
      // On Linux, extract the window ID from the buffer (first 4 bytes, little-endian)
      const windowId = process.platform === 'linux'
      ? hwndBuffer.readUInt32LE(0)
      : hwndBuffer;

      dragAddon.startDrag(windowId);
    } catch (error) {
      console.error(error);
    }
  });
});
```
